conbench/tests/app/test_results.py

```diff
@@ -36,8 +39,46 @@ def test_legacy_route(self, client):
             r"unknown benchmark result ID: \w+", response.text, flags=re.ASCII
         )
 
+    @pytest.mark.parametrize(
+        "samples",
+        [(1,), (3, 5), (3, 5, 7)],
+    )
+    def test_display_result_no_mean(self, client, samples):
+        # Test that benchmark result view/page loads fine for the results that
+        # have one or two multisamples only, i.e. all aggregates are undefined.
+        self.authenticate(client)
+
+        result = _fixtures.VALID_RESULT_PAYLOAD.copy()
+        result["stats"] = {
+            "data": samples,
+            "times": [],  # key must be there as of now, validate more, change this.
+            "unit": "s",
+            "time_unit": "s",
+            # also see https://github.com/conbench/conbench/issues/813
+            # https://github.com/conbench/conbench/issues/533
+            "iterations": len(samples),
+        }
+
+        resp = client.post("/api/benchmark-results/", json=result)
+        assert resp.status_code == 201, resp.text
+        bmr_id = resp.json["id"]
+        resp = client.get(f"benchmark-results/{bmr_id}/")
+        assert resp.status_code == 200, resp.text
+
+    def test_display_result_no_mean_in_history(self, client):
+        # Main goal of this test is to reproduce
+        # https://github.com/conbench/conbench/issues/1155
+        # before testing
+        # https://github.com/conbench/conbench/pull/1167
+        self.authenticate(client)
+        _, bmresults = _fixtures.gen_fake_data(one_sample_no_mean=True)
```

Comment:
should rename this `_fixtures.gen_fake_data` to be more expressive. This is the only entry point I think I have seen in the test suite that attempts to generate a set of benchmark results that are related to each other code history-wise.

Reply:
Yeah, I realized I had to create this fixture when thoroughly testing the lookback z-score functionality